### PR TITLE
Issue/6372 run pyupgrade python39

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description

Run `pyupgrade --py39-plus`

Part of https://github.com/inmanta/inmanta-core/issues/6372

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: tests/data/compile_138/libs/lma/plugins/__init__.py [hunk 1]
```diff
@@ -16,7 +16,6 @@
     Contact: code@inmanta.com
 """
 
-from inmanta import ast
```
Comment (Hugo-Inmanta) [inline tests/data/compile_138/libs/lma/plugins/__init__.py:19]: Might not be ok to remove

File: tests/data/modules_v2/elaboratev2module_extension/inmanta_plugins/elaboratev2module_extension/__init__.py [hunk 2]
```diff
@@ -1 +0,0 @@
-import inmanta_plugins.elaboratev2module
```
Comment (Hugo-Inmanta) [inline tests/data/modules_v2/elaboratev2module_extension/inmanta_plugins/elaboratev2module_extension/__init__.py:1]: same

**Comment by bartv** [comment]
Shouldn't we also run this in the linting phase once we are up to date?

**Comment by bartv** [comment]
Based on the failed tests I also think we can only do this once we are using typeadapter instead of our own typing logic in protocol (cc @sanderr )

**Review by sanderr — CHANGES_REQUESTED** [review]
@Hugo-Inmanta this is going to be merge hell with the pydantic PR, that's why I put that one as a blocker. I think it's easier by far to just run this tool again once that PR is in. If you want to go ahead with the rest of the ticket you can split this task off into a separate ticket.

> Shouldn't we also run this in the linting phase once we are up to date?

I like that.

**Comment by Hugo-Inmanta** [comment]
>  that's why I put that one as a blocker

Damn it, I completely missed the blockers.

**Review by sanderr — COMMENTED** [review]
Wow, this changed way more lines than I would've thought...

**Comment by bartv** [comment]
@sanderr this wil only work if our protocol layer starts supporting dict and list instead of Dict and List. I do not think that is the case and TypeAdapter will solve that for us.

**Pull request closed without merging**